src/app/src/components/MeetTheFishModal.js

```diff
@@ -0,0 +1,212 @@
+    border-radius: 50%;
+    position: absolute;
+    top: 10px;
+    right: 10px;
+    height: 40px;
+    width: 40px;
+    border: 2px solid lightgrey;
+    background: grey;
+    color: white;
+`;
+
+const FishPicture = styled(Box)`
+    background: ${themeGet('colors.lighttan')};
+    border-radius: 50%;
+    height: 400px;
+    width: 400px;
+    position: absolute;
+    top: 50%;
+    transform: translateY(-50%);
+    right: 30px;
+
+    img {
+        height: 350px;
+        position: absolute;
+        left: 50%;
+        top: 50%;
+        transform: translateY(-50%) translateX(-50%);
+    }
+`;
+
+const StyledHeading = styled(Heading)`
+    color: ${themeGet('colors.darkbrown')};
+`;
+
+const StyledText = styled(Text)`
+    color: ${themeGet('colors.darkbrown')};
+`;
+
+const Content = styled(Box)`
+    flex: auto;
+    position: relative;
+    overflow-y: auto;
+    padding: 20px;
+
+    &::-webkit-scrollbar {
+        display: none;
+    }
+`;
+
+export default class MeetTheFishModal extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            index: this.props.index,
+        };
+
+        this.overviewRef = React.createRef();
+        this.characteristicsRef = React.createRef();
+        this.habitatRef = React.createRef();
+        this.lifespanRef = React.createRef();
+    }
+
+    scrollTo(ref) {
+        ref.current.scrollIntoView({ behavior: 'smooth' });
+    }
+
+    getNavigationButton(index) {
+        return (
+            <Button onClick={() => this.setState({ index: index })}>
+                {FISH[index].commonName}
+            </Button>
+        );
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.index !== this.state.index) {
+            this.overviewRef.current.scrollIntoView();
+        }
+    }
+
+    render() {
+        const { children } = this.props;
+
+        const fish = FISH[this.state.index];
+
+        let navButtons = null;
+        if (this.state.index > 0 && this.state.index < FISH.length - 1) {
+            navButtons = (
+                <Flex justifyContent='space-between'>
+                    {this.getNavigationButton(this.state.index - 1)}
+                    {this.getNavigationButton(this.state.index + 1)}
+                </Flex>
+            );
+        } else if (this.state.index === 0) {
+            navButtons = (
+                <Flex justifyContent='flex-end'>
+                    {this.getNavigationButton(this.state.index + 1)}
+                </Flex>
+            );
+        } else {
+            navButtons = (
+                <Flex justifyContent='flex-start'>
+                    {this.getNavigationButton(this.state.index - 1)}
+                </Flex>
+            );
+        }
+
+        return (
+            <StyledPopup
+                trigger={children}
+                modal
+                closeOnDocumentClick
+                onClose={() => this.setState({ index: this.props.index })}
```

Comment:
Why are we resetting the modal index on close? Currently, this errors and breaks the app if we close the modal on page 2+: this is because the modal gets destroyed before `componentDidUpdate` gets called from the state change, and it is confused because the `overviewRef` doesn't exist.

I don't think we need to reset the index state? It should always come in as a `prop` when the modal is opened because we'll be clicking on a fish. For now we should be able to send "0" index in fresh every time we open the modal. The modal will probably need some rearranging.

Reply:
Ah hmm, this is _needed_ in development, because the modal _doesn't_ get destroyed when closed, and so without resetting the state it won't restart on the right index, but that clearly isn't the case in the production build.

I added a guard to the call to `scrollIntoView` to check that the `current` element isn't null in https://github.com/azavea/pwd-fishway/pull/69/commits/466bdde09a93f57578236706a0733770c58aa449